cpp/src/io/utilities/column_utils.cuh

```diff
@@ -57,27 +57,24 @@ rmm::device_uvector<column_device_view> create_leaf_column_device_views(
   auto leaf_columns = cudf::device_span<column_device_view>{leaf_column_views};
```

Comment:
I feel like this function should also be more explicit about expecting the incoming data to be parquet_column_views.  Maybe pass a vector of parquet_column_views instead of the table?  Would that end up jumbling the code up in write() too much?

Reply:
There are two reasons why we should not make this function deal with parquet_column_views :

1. This function is meant to be used by both ORC and parquet writer.

2. The point of this function is to reduce the involvement of host in initialization of members of `ColumnDescriptor col_desc`. As of now `ColumnDescriptor` can either be `EncColumnDesc` in parquet writer or `stats_column_desc` in orc writer. The reasoning behind why this function was added is documented in [issue 6893](https://github.com/rapidsai/cudf/issues/6893) :

> The construction of all these column_device_views might cause many cudaMemcpys if done using column_device_view::create. Since write_parquet and write_chunk are called with a table_view, it'd be better to call table_device_view::create and use a kernel to copy them into EncColumnDesc.

In short, we need `column_device_view`s associated with each column to be used at a later point. This was created using `table_device_view::create()` which is faster than iterating on host and calling functions to create `column_device_view` from a single column. Since we are also dealing with lists we need a parent `column_device_view` and a leaf `column_device_view` and this is what this function deals with and this is done on the device.

Doing this via a host vector of `parquet_column_view` would likely involve data transfers we don't need.